### DISCUSSION TITLE
How can my extension trigger a refresh of the Source Control Graph after using the git extension's API to fetch changes?

### DISCUSSION THREAD

**Discussion by gjsjohnmurray** [microsoft/vscode-discussions#1608] [Extension Development QnA] [answered]
Here's the code fragment from the extension which I am testing with 1.94 Insiders:

```ts
    if (remoteName) {
        await this.git?.getRepository(uri)?.fetch(remoteName);
        vscode.commands.executeCommand('workbench.scm.action.refreshGraph');
    }
```

I am convinced that the `workbench.scm.action.refreshGraph` command executes because if I scroll down in the graph before triggering the code in my extension I see the graph scroll back to its top line. But the refresh doesn't actually occur until I manually click the Refresh button.

My hunch is that the command only knows which repository's graph is being displayed (and thus should be refreshed from) when I click the button. If that's true, is there a way I can pass that information as a command argument? Or can the command be made smart enough to handle this case?

Pinging @lszomoru

**Comment by lszomoru** [comment, marked as answer]
The refresh command is not really meant to be used by extensions (note that as part of cleaning up things, the command has been also been renamed to `workbench.scm.action.graph.refresh`). When invoking the command, it will always refresh the graph for the repository that is rendered in the Source Control Graph view.

**Reply by gjsjohnmurray** [reply]
Thanks for responding. Latest Insiders seems to be automatically refreshing the graph after a fetch, so I no longer need to trigger a manual refresh.

**Reply by lszomoru** [reply]
Refreshing the graph following a `fetch` operation is tricky since a fetch can occur as part of a manual action, or in the background as part of auto-fetch. At the moment we only refresh the graph if it is scrolled to the top as we do not want to refresh it while the user is navigating it. If the graph is not scrolled to the top, we will display an `Outdated` badge in the view title to indicate to the user that data could be missing from the graph. Hope that makes sense.
